File: src/main/java/nz/co/searchwellington/model/SiteInformation.java
package nz.co.searchwellington.model;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class SiteInformation {
	
    @Value("${areaname}")
	private String areaname;

    @Value("${installed}")
	private String url;
    
    @Value("${twitter.username}")
    private String twitterUsername;

    @Value("${imageroot}")
    private String imageRoot;
    
    @Value("${staticroot}")
    private String staticRoot;
    
    public SiteInformation() {
	}
        
    public String getAreaname() {
		return areaname;
	}
    
	public String getSitename() {
	    return "Search " + this.areaname;
	}
	
	public String getTagline() {
		return areaname + " in a box";
	}
	
	public String getUrl() {
		return url;
	}
	
    public String getTwitterUsername() {
        return twitterUsername;
    }

	public String getStaticRoot() {
		return staticRoot;
	}
	
	public String getImageRoot() {
		return imageRoot;
	}

	public boolean isTwitterEnabled() {
		return twitterUsername != null && !twitterUsername.isEmpty();
	}
		
}

File: src/main/scala/nz/co/searchwellington/model/SiteInformation.scala
package nz.co.searchwellington.model

import org.springframework.beans.factory.annotation.{Autowired, Value}
import org.springframework.stereotype.Component

@Component
class SiteInformation @Autowired()(@Value("${areaname}") areaname: String = "",
                                   @Value("${installed}") url: String = "",
                                   @Value("${twitter.username}") twitterUsername: String = "",
                                   @Value("${imageroot}") imageRoot: String = "",
                                   @Value("${staticroot}") staticRoot: String = "",
                                   @Value("${sitename}") sitename: String = "") {

  def getAreaname: String = areaname

  def getSitename: String = sitename

  def getUserAgent: String = "wellynews (" + url + "/about)"

  def getTagline: String = areaname + " in a box"

  def getUrl: String = url

  def getTwitterUsername: String = twitterUsername

  def getStaticRoot: String = staticRoot

  def getImageRoot: String = imageRoot

}
